packages/block-library/src/heading/deprecated.js

```diff
@@ -30,17 +31,77 @@ const blockAttributes = {
 	placeholder: {
 		type: 'string',
 	},
-	textColor: {
-		type: 'string',
-	},
-	customTextColor: {
-		type: 'string',
-	},
+};
+
+const migrateCustomColors = ( attributes ) => {
+	if ( ! attributes.customTextColor ) {
+		return attributes;
+	}
+	const style = {
+		color: {
+			text: attributes.customTextColor,
+		},
+	};
+	return {
+		...omit( attributes, [ 'customTextColor' ] ),
+		style,
+	};
 };
 
 const deprecated = [
 	{
-		attributes: blockAttributes,
+		supports: blockSupports,
```

Comment:
I forgot about one thing, it would be nice to add the test case that covers this new deprecation:
- a block that has `has-text-color` class set?

@jorgefilipecosta should know best what's the snippet that makes this deprecation unique.

Reply:
I will look into the deprecations and try to add a snippet specific for this deprecation 👍

Reply:
Thanks @jorgefilipecosta 🙇